### DISCUSSION TITLE
Getting the ref for a model instance

### DISCUSSION THREAD

**Discussion by knownasilya** [mikkopaderes/ember-cloud-firestore-adapter#266] [Q&A]
Is there a way to get the ref for a model? Like:

```ts
const model = this.store.findRecord('post', { buildReference...})
const ref = getRef(model);
```

---

Kinda asking because I'm getting a timeout when I set the same model on itself as a belongs to:

```ts
 const post = this.dataStore.createRecord('post', {
  repost: this.poster.repostPost, // <---- this
  //...
});

await post.save({
  adapterOptions: {
    buildReference: (db: Firestore) => {
      return collection(
        db,
        `accounts/${this.args.account.id}/feeds/${this.args.feed.id}/posts`
      );
    },
  },
});
```

Where the `post` model has `@belongsTo('post') repost!: Promise<Post>`. On save the promise times out and the tab is shot.

**Comment by mikkopaderes** [comment]
No support for that right now. Firestore being a NoSQL database, may tend to have duplicate data all over to improve query performance. Having a way to get the ref of a model would be difficult when that happens.

As for the timeout, I'm not sure why that happens for your case. On my end, I don't have any problem with having `belongsTo` that points to itself.
